### DISCUSSION TITLE
Interested to add others plugins to that project ?

### DISCUSSION THREAD

**Discussion by nhuray** [gperdomor/oss#713] [General]
Hi @gperdomor 

First I would like to thank you for that project. I'm using extensively the `nx-container` executor in my projects. 

Because in my company we are deploying to Kubernetes I started to create couple of Nx executors for the Kubernetes ecosystem. 

I would like to share those executors with the community so I though to start an open source project. But what's painful at the moment is the profusion of Nx plugins all over Github. Moreover, some of the executors have dependency to `@nx-tools/container-metadata`. So I though it might be better to publish those executors and generators in your project to have more visibility and support on the long term.

Most of the time those executors are just wrapper around the command line (e.g kustomize) but sometimes there's a bit of logic.

So at the moment this is the list of executors I developed:
  - `kustomize`: Build kubernetes manifests with [kustomize](https://kustomize.io/)
  - `kubectl`: Run `kubectl apply` 
  - `k3d`: Create local Kubernetes cluster with K3D
  - `gitops`: Executor editing a Kustomize app in GitHub (only) 

I also create a generator to create a typical `kustomize` project structure.

Please let me know what do you think.

Regards,

Nicolas

**Comment by nhuray** [comment]
@gperdomor any thoughts ?

**Reply by gperdomor** [reply]
:o I don't see any problem, if you want to contribute with that I would be more than happy ❤️

**Reply by nhuray** [reply]
Cool stuff ! 

I'm gonna start working on it when I have a little time. 

This is the plugin organization I envision:

```
.
packages
├── nx-kubernetes
│   ├── src
│   │   ├── executors
│   │   │   ├── build                    # Executors to build kubernetes manifests (support Helm and Kustomize)
│   │   │   ├── deploy                   # Executors to apply kubernetes manifests (support kubectl and gitops)
│   │   │   └── provision                # Executors to provision local Kubernetes cluster (support kind and k3d)
│   │   ├── generators
│   │   │   └── init                     # Generators to create a Kubernetes app (support Helm chart and Kustomize)
```

The idea is to be implement something like you did with the `engine` in `nx-container`, to be able to choose an implementation as an executor option (e.g `helm` or `kustomize` for the `build` executor).

Along the way community can add more implementation (e.g [kpt](https://kpt.dev/) or [cdk8s](https://cdk8s.io/) )

Thanks again for you project !
